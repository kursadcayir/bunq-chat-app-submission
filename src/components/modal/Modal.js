
import './modal.css'

const Modal = (props) => {

    const {
        children,
        onClose,
        title,
        isOpen,
        className,
        ...otherProps
    } = props;
    return (
        <div className={`${isOpen ? "modal display-block" : "modal display-none"} ${className}`} {...otherProps}>
        <div className="modal-main">
            <header className="modal-card-head d-flex align-items-center">
            <p className="modal-card-title">{title}</p>
            <button className="btn btn-light delete" aria-label="close" onClick={onClose}> Close</button>
            </header>
            <section className="modal-card-body">
            {children}
            </section>
        </div>
        </div>
    );
}
export default Modal;