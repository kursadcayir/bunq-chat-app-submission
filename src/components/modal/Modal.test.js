import { render,screen } from "@testing-library/react"
import Modal from './Modal'

it( "The Modal SnapShot", () => {
    let testMessage = "THIS IS TEST CASE";
    render( <Modal isOpen={true} className={'test'} onClose={ () => {console.log("test")}}>  {testMessage} </Modal>  );
    expect(screen.getByText(testMessage)).toMatchSnapshot();
    expect(screen.getByText(testMessage)).toBeInTheDocument();

} )