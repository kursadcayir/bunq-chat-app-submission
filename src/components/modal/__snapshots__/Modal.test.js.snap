// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Modal SnapShot 1`] = `
<section
  class="modal-card-body"
>
    
  THIS IS TEST CASE
   
</section>
`;
